crates/primitives-traits/src/tx_type.rs

```diff
@@ -0,0 +1,28 @@
+use alloy_eips::eip2718::Eip2718Error;
+use alloy_primitives::{U64, U8};
+use alloy_rlp::{Decodable, Encodable};
+use core::fmt::{Debug, Display};
+
+/// Trait representing the behavior of a transaction type.
+pub trait TxType:
+    Into<u8>
+    + Into<U8>
+    + PartialEq
+    + Eq
+    + PartialEq<u8>
+    + TryFrom<u8, Error = Eip2718Error>
```

Comment:
do we need the `Error = ` specification here?

Reply:
yes